### PR TITLE
Simplify the parameter for forward index creators

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Also move method getNumBitsPerValue() into PinotDataBitSet and re-implement it without using double

### CODE REVIEW

File: pinot-core/src/test/java/com/linkedin/pinot/core/io/util/PinotDataBitSetTest.java [hunk 1]
```diff
@@ -27,6 +27,25 @@
   private static final Random RANDOM = new Random();
   private static final int NUM_ITERATIONS = 1000;
 
+  @Test
+  public void testGetNumBitsPerValue() {
+    for (int i = 0; i < NUM_ITERATIONS; i++) {
+      int cardinality = RANDOM.nextInt(Integer.MAX_VALUE) + 1;
+      int actual = PinotDataBitSet.getNumBitsPerValue(cardinality);
+      if (cardinality == 1) {
```
Comment (snleee) [inline pinot-core/src/test/java/com/linkedin/pinot/core/io/util/PinotDataBitSetTest.java]: The probability of random picking `cardinatlity = 1` is too low. Maybe it's better to test this separately outside of this loop since it's an edge case?

Comment (snleee) [inline pinot-core/src/test/java/com/linkedin/pinot/core/io/util/PinotDataBitSetTest.java]: Since this is an edge case, can we check separately when cardinality = 1? Currently, it will hit 1 with very very low chance.

File: pinot-core/src/main/java/com/linkedin/pinot/core/io/util/PinotDataBitSet.java [hunk 2]
```diff
@@ -39,6 +39,20 @@
     }
   }
 
+  public static int getNumBitsPerValue(int cardinality) {
```
Comment (snleee) [inline pinot-core/src/main/java/com/linkedin/pinot/core/io/util/PinotDataBitSet.java]: Can you put some comments on how we decide the number of bits per value in high level with some simple examples?

File: pinot-core/src/test/java/com/linkedin/pinot/core/io/util/PinotDataBitSetTest.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-core/src/test/java/com/linkedin/pinot/core/io/util/PinotDataBitSetTest.java]: Good point. Added

File: pinot-core/src/main/java/com/linkedin/pinot/core/io/util/PinotDataBitSet.java (hunk 2 above)
Comment (Jackie-Jiang) [reply pinot-core/src/main/java/com/linkedin/pinot/core/io/util/PinotDataBitSet.java]: Added

File: pinot-core/src/main/java/com/linkedin/pinot/core/io/util/PinotDataBitSet.java [hunk 3]
```diff
@@ -39,6 +39,20 @@
...
+    // Use at least one bit even there is only one possible value
```
[the remark this answers is not in the document]
Comment (Jackie-Jiang) [reply pinot-core/src/main/java/com/linkedin/pinot/core/io/util/PinotDataBitSet.java]: There is no **Math.log2()**. In order to do so, we have to do **Math.ceil(Math.log(cardinality) / Math.log(2))**, while I am afraid that using double division could cause wrong result because of the precision error.
As for the **Math.max()**, we can avoid the actual calculation by using the if condition check.
